osfoffline/sync/ext/watchdog.py

```diff
@@ -1,242 +1,83 @@
-                            return
-
-            if event.event_type == EVENT_TYPE_MODIFIED:
-                if event.is_directory:
-                    return
-                for evt in self._event_cache.children():
-                    if evt.event_type == EVENT_TYPE_MOVED and evt.dest_path == event.src_path:
-                        return
-
-            if event.event_type == EVENT_TYPE_CREATED:
-                if (not event.is_directory and parts in self._event_cache and self._event_cache[parts].event_type == EVENT_TYPE_DELETED):
-                    Event = DirModifiedEvent if event.is_directory else FileModifiedEvent
-                    self._event_cache[parts] = Event(event.src_path)
-                else:
-                    logger.debug('inside create_event:')
-                    logger.debug(self._event_cache.children())
-                    delete_events = [
-                        evt
-                        for evt in self._event_cache.children()
-                        if evt.event_type == EVENT_TYPE_DELETED
-                        and evt.basename == event.basename
-                        and evt.sha256 == event.sha256
-                    ]
-                    if delete_events:
-                        for evt in delete_events:
-                            # delete the delete
-                            del self._event_cache[evt.parts]
-                            Event = DirMovedEvent if event.is_directory else FileMovedEvent
-                            self._event_cache[evt.parts] = Event(src_path=evt.src_path, dest_path=event.src_path)
-                    else:
-                        move_events = [
-                            evt
-                            for evt in self._event_cache.children()
-                            if evt.event_type == EVENT_TYPE_MOVED
-                        ]
-                        if move_events:
-                            for evt in move_events:
-                                if evt.dest_path in event.src_path:
-                                    return
-                        self._create_cache.append(event)
-
-            else:
-                if not consolidate and parts in self._event_cache:
-                    ev = self._event_cache[parts]
-                    if not isinstance(ev, OrderedDict) and ev.event_type == EVENT_TYPE_DELETED:
-                        # For leaf entries, turn deletes followed by creates into updates,
-                        #   eg saving in vim or replacing a file in finder.
-                        Event = DirModifiedEvent if event.is_directory else FileModifiedEvent
-                        event = Event(event.src_path)
-                self._event_cache[parts] = event
-
-            logger.debug('Create cache: {}'.format(self._create_cache))
-            logger.debug('Event cache: {}'.format(self._event_cache))
+            try:
+                event.sha256 = sha256_from_event(event)
+            except FileNotFoundError:
```

Comment:
Might be good to move this to `sha_256_from_event` (other error handling is already happening there)